conformance/utils/suite/suite.go

```diff
@@ -374,6 +379,39 @@ func (suite *ConformanceTestSuite) Setup(t *testing.T, tests []ConformanceTest)
 	}
 }
 
+func (suite *ConformanceTestSuite) setClientsetForTest(test ConformanceTest) error {
+	cfg, err := clicfg.GetConfig()
+	if err != nil {
+		return err
+	}
+	featureNames := []string{}
+	for _, v := range test.Features {
+		featureNames = append(featureNames, string(v))
+	}
+	if len(test.Features) == 0 {
+		featureNames = []string{"unknownFeature"}
+	}
+	cfg.UserAgent = strings.Join(
+		[]string{
+			testSuiteUserAgentPrefix,
+			suite.apiVersion,
+			test.ShortName,
+			strings.Join(featureNames, ","),
```

Comment:
Seems like this list could get pretty long, is it okay to have all the feature names present in every useragent string? How long can a useragent string be?

Reply:
@youngnick, good question!

Looking at the latest Kubernetes test results, filtering by conformance tests:
```bash
$ curl -sSL https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce/1810176656531263488/artifacts/junit_01.xml \ 
  | yq -p=xml -o=yaml \
  | yq '.testsuites.testsuite.testcase[] | select(.+@name == "*[Conformance]*") | .+@name' \
  | sed 's/.*\]\(.*\)\[.*/\1/g' \
  | awk '{ print length($0) " " $0; }' | sort -r -n | cut -d ' ' -f 2- | head -n 1 | wc -c
```

The longest useragent is 131 characters.
The current longest useragent with the changes from this PR is 
```
apisnoop=# select distinct(useragent) as ua, length(useragent) as ual from testing.audit_event where useragent ilike 'gateway-api-conformance.test%' order by ual desc limit 1;
                                                                         ua                                                                          |
 ual
-----------------------------------------------------------------------------------------------------------------------------------------------------+
-----
 gateway-api-conformance.test::v1.2.0-dev::HTTPRouteRequestMultipleMirrors::Gateway,HTTPRoute,HTTPRouteRequestMirror,HTTPRouteRequestMultipleMirrors |
 147
(1 row)
```

is 147.

Kubernetes Audit Logging appears to only store 1024 characters for useragents
https://github.com/kubernetes/kubernetes/blob/2b03f04/staging/src/k8s.io/apiserver/pkg/audit/request.go#L39

Reply:
Thanks for the detailed analysis @BobyMCbobs! If I'm understanding this correctly, it seems like this would be easy to change in the future if/when we run into an issue, and the only limit that really matters is k8s API Server. Given that, I think I'm ok with the current path. In the future we could add a numerical id for each feature if needed.

Reply:
@robscott, absolutely!
Yes and the main thing is that we are able to map tests to features. Using this as a framework, we can pump any extra data through as usefulness and need arises.

Reply:
> In the future we could add a numerical id for each feature if needed.

I think this is a great idea, instead of putting all the features' names here.